### DISCUSSION TITLE
How to create new tools?

### DISCUSSION THREAD

**Discussion by birchcode** [TransformerOptimus/SuperAGI#295] [Q&A]
I've created an agent to build an app with next.js and so it needs to run terminal to perform commands, but we don't have a Terminal tool. 

![Screenshot 2023-06-10 at 06 19 07](https://github.com/TransformerOptimus/SuperAGI/assets/1013599/c9c0bb7c-1014-4385-9c0b-7d01f58d075f)

I can see in agent_executor.py we have a list of tools being imported. Should I be attempting to create my own Terminal module in superagi/tools or do we have a directory somewhere of additional tools like this to use?

**Comment by progone** [comment]
I've created some tools. They drop into the output folder of your directory.  Getting them to be used is another story.  I believe some of the tools my tool agent made got it to move around locally on a macbook.  First, it took a while in the output directory (folder) before going the proper direction (upward).  It found its way into Desktop then into the test folder, read test.txt and finished with complete.   It take a lot of time thou.  Perhaps next time I ask Agents to use tools in output DIR, and explain the output DIR has no sub directories and not to create any, unless its testing mkdir command within output DIR.   Just sharing my experience on day 3.

**Reply by birchcode** [reply]
Thanks for chiming in. I wondered if the Agent is supposed to build its own tools or not.

**Comment by neelayan7** [comment]
Hey, in today's release, we've built an easy way where people can add their tools simply by making a Github Repo with all the logic and attaching the link in SuperAGI's front-end. I hope that helps you.

**Reply by markesphere** [reply]
Using the interface, I tried adding a PyMed tool to access the pubmed API and tried the "add new tool"with the repo https://github.com/PyMed/PyMed.git and it gave me a an "Invalid Github URL".   Are there special requirements for the repo url?
